dcm4che-imageio/src/main/java/org/dcm4che3/imageio/plugins/dcm/DicomImageReader.java

```diff
@@ -587,22 +618,83 @@ private void readMetadata() throws IOException {
                 pixelDataLength = dis.length();
                 if (pixelDataLength == -1)
                     epdiis = new EncapsulatedPixelDataImageInputStream(dis, ds.getInt(Tag.NumberOfFrames, 1));
+            } else {
```

Comment:
I will have another review/issue to handle the rest of the post pixel data changes, but this example handles it if the object happens to not be an image containing object.